### DISCUSSION TITLE
Typer captures `EOFError` and shallows the traceback.

### DISCUSSION THREAD

**Discussion by pmav99** [fastapi/typer#628] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


def main():
    # raise Exception("Some Error")  # if you uncomment this line, you get a traceback.
    raise EOFError("Some Error")   # No traceback. Just "Aborted" printed on stderr


if __name__ == "__main__":
    typer.run(main)
```

### Description

Typer seems to be capturing `EOFError` and consequently code that raises an `EOFError` on its own doesn't print a traceback, just the "Aborted" message.

The problem seems to be these lines: https://github.com/tiangolo/typer/blob/f83c66d539b95d00de74e7de6a14b5c53ae0bfe6/typer/core.py#L227-L229

As soon as you realize that is typer that shallows the exception, it is not hard to find a workaround. For example you can capture the `EOFError` and raise a different exception:

``` python
def main():
    try:
        raise EOFError("some problem")
    except EOFError as exc:
        raise RuntimeError("Some message") from exc
```

Nevertheless, it would be nice if `EOFError` could be handled normally. 

Question, is it truly necessary to bundle `EOFError` with `KeyboardException`?

### Operating System

Linux

### Typer Version

0.9.0

### Python Version

3.11

**Comment by NiklasRosenstein** [comment]
I just ran into the same issue and it was incredibly frustrating to debug.

**Reply by riklopfer** [reply]
Just ran into this when attempting to unpack a corrupt archive using the python standard library. All I see on in the console is "Aborted"

**Comment by fepegar** [comment]
@pmav99 Can you create an issue instead? This shouldn't be a discussion.

**Reply by svlandeg** [reply]
Issues are typically only created by Tiangolo or other maintainers.
